### PR TITLE
python310Packages.napari-npe2: 0.5.1 -> 0.6.1

### PR DESCRIPTION
**Pull request by onny**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fabaff — COMMENTED** [review]
This module only supports [>=3.8](https://github.com/napari/npe2/blob/v0.6.1/setup.cfg#L34).

**Comment by onny** [comment]
Thank you @fabaff for reviewing

File: pkgs/development/python-modules/napari-npe2/default.nix [hunk 1]
```diff
@@ -13,11 +13,14 @@
 , rich
 , setuptools-scm
 , typer
+, build
+, rich
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/napari-npe2/default.nix:17]:
```suggestion
```

Comment (mweinelt) [reply pkgs/development/python-modules/napari-npe2/default.nix:17]: Duplicates break eval.

Comment (onny) [reply pkgs/development/python-modules/napari-npe2/default.nix:17]: When I remove these build inputs I get following error
```
       last 10 log lines:
       > buildPhase completed in 36 seconds
       > installing
       > Executing pipInstallPhase
       > /build/source/dist /build/source
       > Processing ./npe2-0.6.1-py2.py3-none-any.whl
       > Requirement already satisfied: pydantic in /nix/store/5m262qkll34vnvjln1ac54rih2b404s7-python3.10-pydantic-1.9.1/lib/python3.10/site-packages (from npe2==0.6.1) (1.9.1)
       > Requirement already satisfied: psygnal>=0.3.0 in /nix/store/0nr6c33yi5kddahr22v4iyk188jlcx49-python3.10-psygnal-0.3.5/lib/python3.10/site-packages (from npe2==0.6.1) (0.3.5)
       > ERROR: Could not find a version that satisfies the requirement build (from npe2) (from versions: none)
       > ERROR: No matching distribution found for build
```

Comment (mweinelt) [reply pkgs/development/python-modules/napari-npe2/default.nix:17]: It is duplicated in the argument list, you're apparently removing it from propagatedBuildInputs.

Comment (onny) [reply pkgs/development/python-modules/napari-npe2/default.nix]: Oh that is obvious, sorry for that

**Review by SomeoneSerge — APPROVED** [review]
Hi, sorry I'm late to the party. I just ran `nix run github:onny/nixpkgs/napari-npe2#napari` and it's OK

**Pull request merged**